examples/server/server.cpp

```diff
@@ -1596,7 +1596,7 @@ struct server_queue {
     }
 
     // Add a new task, but defer until one slot is available
-    void defer(server_task task) {
+    void defer(server_task & task) {
```

Comment:
Wouldn't it be better to use r-value reference so it is clearer at the caller site that we transfer the ownership:

```suggestion
    void defer(server_task && task) {
```

Reply:
Thanks, that's a good idea, I implemented this in https://github.com/ggml-org/llama.cpp/pull/12936/commits/b7aea032eb85725ca4150bbc8963b88088ba8b1d